File: app/src/main/java/com/zhxh/rxjava2/samples/model/Car.java
package com.zhxh.rxjava2.samples.model;

import java.util.concurrent.Callable;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;

/**
 * Created by zhxh on 30/08/16.
 */
public class Car {

    private String brand;

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public Observable<String> brandDeferObservable() {
        return Observable.defer(new Callable<ObservableSource<? extends String>>() {
            @Override
            public ObservableSource<? extends String> call() throws Exception {
                return Observable.just(brand);
            }
        });
    }

}

File: app/src/main/java/com/zhxh/rxjava2/samples/model/Car.kt
package com.zhxh.rxjava2.samples.model

import java.util.concurrent.Callable

import io.reactivex.Observable
import io.reactivex.ObservableSource

/**
 * Created by zhxh on 30/08/16.
 */
class Car {

    private var brand: String? = null

    fun setBrand(brand: String) {
        this.brand = brand
    }

    fun brandDeferObservable(): Observable<String> {
        return Observable.defer { Observable.just(brand!!) }
    }

}
